disnake/client.py

```diff
@@ -2490,7 +2494,16 @@ async def bulk_overwrite_global_commands(
         """
         for cmd in application_commands:
             cmd.localize(self.i18n)
-        return await self._connection.bulk_overwrite_global_commands(application_commands)
+        res = await self._connection.bulk_overwrite_global_commands(application_commands)
+
+        for api_command in res:
+            cmd = utils.get(application_commands, name=api_command.name)
```

Comment:
It's possible to have 2 app commands of different types (or guilds) with identical names, so this should be modified